### PR TITLE
Microhapulator

### PR DESCRIPTION
**Pull request by standage**
New recipe for MicroHapulator 0.1.

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by standage** [comment]
Hi @bioconda/python, is there any way to mix bioconda dependencies and pip dependencies? I'm pretty sure that's the issue with my build failure: some of the packages this depends on are in bioconda, but some are only in PyPI.

**Comment by standage** [comment]
InSilicoSeq is the only dependency not in conda/bioconda. I just opened up #15142 to address that.

**Comment by epruesse** [comment]
@standage In theory, `conda` can install `pip` dependencies. In practice, we don't allow that. It can't be controlled well. Among other things, PyPI does not have dependency meta data. Each `setup.py` get's to decide on those on the fly based on code. So it's a feature from the early days that I'd consider deprecated for good reason.

**Comment by standage** [comment]
Thanks @epruesse. I figured it would be better to have it in bioconda anyway. #15240

File: recipes/microhapulator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  script: python -m pip install --no-deps --ignore-installed .
+  number: 0
+
+requirements:
+  host:
+    - python >=3
+    - pip
+  run:
+    - happer >=0.1
+    - insilicoseq >=1.3.5,!=1.3.5-py_0
```
Comment (standage) [inline recipes/microhapulator/meta.yaml]: Yeah, I tried something like this earlier and also failed the linter.

Comment (epruesse) [reply recipes/microhapulator/meta.yaml]: It's not the linter... it just breaks in `conda`.

Comment (epruesse) [reply recipes/microhapulator/meta.yaml]: Sorry, I forgot about this one.

**Comment by standage** [comment]
The problematic package just pushed a new bugfix release (in #15353). I'm going to give that several minutes to propagate through the system, and then push an update to this package's `meta.yaml` to depend on this new version.

That doesn't solve the problem that's been halting progress on this thread, but hopefully it should allow this package to finally build successfully.

**Comment by standage** [comment]
AWESOME! Now I'm getting issues with the package tests. I can *do* something about that! 😀

**Comment by standage** [comment]
Good grief, I'm glad the build is finally passing. Ready for review and merge!

**Comment by epruesse** [comment]
Great! That sidesteps the issue!

Elmar

--
message sent from mobile device

File: recipes/microhapulator/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - python >=3
+    - pip
+  run:
+    - happer >=0.1
+    - insilicoseq >=1.3.6
+    - microhapdb >=0.3
+    - numpy >=1.15.4
+    - python >=3
+    - pyfaidx >=0.5.5.2
+    - pysam >=0.15.2
```
Comment (epruesse) [inline recipes/microhapulator/meta.yaml]: Want to pin that one to `>=0.15.2,<0.16.0a0`? I think they changed API before on patch level bumps.

File: recipes/microhapulator/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+test:
+  imports:
+    - microhapulator
+  requires:
+    - pytest >=3.10
+    - pytest-cov >=2.6
+    - wget
```
Comment (epruesse) [inline recipes/microhapulator/meta.yaml]: This was generated I think. Do you need the requires? 

Is there some other useful test you can run, like a command? Nothing big, just something to test whether it looks like it should in principle work.

File: recipes/microhapulator/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+wget -O hg38.fasta.gz http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz
+gunzip hg38.fasta.gz
+faidx hg38.fasta chr1:100-150 > /dev/null
+
+pytest -m 'not known_failing' --cov=microhapulator --pyargs microhapulator
```
Comment (epruesse) [inline recipes/microhapulator/run_test.sh]: Ah, never mind, you are running the pytests. That's a lot more testing than a lot of our recipes see. We really need a  way to store the test data though. Hg38 will be pretty big. Let's hope the link never goes away.

**Comment by epruesse** [comment]
@biocondabot merge

File: recipes/microhapulator/meta.yaml (hunk 2 above)
Comment (standage) [reply recipes/microhapulator/meta.yaml]: Good to know. I'll consider that for the next update.

File: recipes/microhapulator/run_test.sh (hunk 4 above)
Comment (standage) [reply recipes/microhapulator/run_test.sh]: These are the same tests that are run as part of MicroHapulator's CI build on Travis. For the next update I'm considering a less comprehensive smoke test just for bioconda purposes.

**Comment by standage** [comment]
Thanks for your feedback @epruesse!

File: recipes/microhapulator/meta.yaml (hunk 2 above)
Comment (standage) [reply recipes/microhapulator/meta.yaml]: Addressed in https://github.com/bioconda/bioconda-recipes/pull/15522/commits/9f0879f8d8205d76eda2a903218182dceaee9a77.

**Pull request merged**